mautic/mautic
# 2.16.0 beta merge to 3.x

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Merging changes form 2.16.2-beta into 3.x.

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Test the whole Mautic basically :) Lots of changes, some merge conflicts could go wrong. I guess we'll use the M3 beta testing to ensure everything works.
